src/components/PositionMarker.js

```diff
@@ -0,0 +1,28 @@
+import { Marker, useMapEvents } from "react-leaflet";
+
+
+const PositionMarker = ({ location, setLocation, markerLocation, setMarkerLocation }) => {
+  useMapEvents({
+    click(e) {
+      if (setLocation && setMarkerLocation) {
```

Comment:
`if (setLocation && setMarkerLocation)`
This looks really strange to me. Is there really a case when you are not sending in the setters?